Managing ssl parameter into query for JetPack.

develric opened the pull request:
Fixes #11249

Some more context in the inked issue above and in `p1584956107248200-slack-ios-osx-dev`

**NOTE:  I do not have full clarity about the correct procedure to final get merged this PR into the original WP-Utils subtree (after review/approve, let's sync before to merge this one).**

My best understanding is the steps reported here `p1570688351009600-slack-android-dev` but the tests update related to the PR here #11524 AFAIU does not seem to be merged into WPUtils develop (cc @jkmassel in case I'm missing something if he can help to clarify 🙇‍♂️)

### To test:
- Check you can access Media gallery images without errors with:
    - WP.com site, 
    - https JetPack site (added by address or by WP.com login), 
    - http JetPack site (added by address or by WP.com login) 
- Check the WP-Utils connected tests pass
- Check you can repro and get fixed the issue the user had in #11249 setting up Charles Proxy with SSP (further details here `p99K0U-ZF-p2`)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

oguzkocer commented:
This issue is targeting the current frozen milestone which is about to be released. I am bumping it to `14.6` which is when a PR that lands to `develop` in the next 2 weeks will be released. If it's more urgent than that, please make sure to target `release/14.5` branch which will be available later today and update the milestone for the issue accordingly.

planarvoid commented on libs/utils/WordPressUtils/src/androidTest/java/org/wordpress/android/util/PhotonUtilsTest.java:
```diff
@@ -0,0 +1,104 @@
+    public void getPhotonImageUrlIsEmptyWhenUrlIsNull() {
+        String photonUrl = PhotonUtils.getPhotonImageUrl(null, 0, 1);
+
+        assertEquals("", photonUrl);
```
I'd rather use `assertThat(photonUrl).isEmpty()` and generally use `assertThat` than other asserts. I think it's more readable and I like the consistency.

develric replied:
Agree. AFAIU in the WPUtils we use JUnit and Hamcrest, so I moved using that assertThat pattern here 9881777ae35ad408285b5452a5639248e79b543d 😊

develric commented:
Hey @planarvoid , thanks for the review! I moved to the assertThat pattern here 9881777ae35ad408285b5452a5639248e79b543d and also changed the lib version here 12b7277b44b8d996bfacaefbf91071b99f22fa16 (for what I now understand I get we need to manage it here as well as in the subtree).

develric commented:
Hey @planarvoid, sorry to ping again on this one 😊! As per an alignmentt chat with @oguz, we decided to keep the version of the Utils Lib here since we are not making a release from this right now (there are some other PRs modifications sync to do also and the release will be managed in a separate task).
